### PR TITLE
nuke-room-from-db.sh: added postgresql option and help

### PR DESCRIPTION
**Pull request by rubo77**
Signed-off-by: Ruben Barkow <github@r.z11.de>

### CODE REVIEW

**Comment by rubo77** [comment]
This still seems to work on postresql, though I get some warnings during that script:

```
FEHLER:  Relation »event_search_content« existiert nicht  
ZEILE 1: DELETE FROM event_search_content WHERE c1room_id = '!gniSuNf...  
                     ^  
DELETE 5
...
  
WARNUNG:  überspringe »pg_authid« --- nur Superuser kann sie vacuumen  
WARNUNG:  überspringe »pg_database« --- nur Superuser kann sie vacuumen  
WARNUNG:  überspringe »pg_db_role_setting« --- nur Superuser kann sie vacuumen  
WARNUNG:  überspringe »pg_tablespace« --- nur Superuser kann sie vacuumen  
WARNUNG:  überspringe »pg_pltemplate« --- nur Superuser kann sie vacuumen  
WARNUNG:  überspringe »pg_auth_members« --- nur Superuser kann sie vacuumen  
WARNUNG:  überspringe »pg_shdepend« --- nur Superuser kann sie vacuumen  
WARNUNG:  überspringe »pg_shdescription« --- nur Superuser kann sie vacuumen  
WARNUNG:  überspringe »pg_replication_origin« --- nur Superuser kann sie vacuumen  
WARNUNG:  überspringe »pg_shseclabel« --- nur Superuser kann sie vacuumen  
...
```

**Comment by rubo77** [comment]
Maybe we should remove the VACUUM at the end or make it optional

this can take some hours on a large database

**Comment by rubo77** [comment]
Also shouldn't the media posted in that room be deleted before you nuke the room?

**Comment by rubo77** [comment]
I think this is still an open bug

**Comment by rubo77** [comment]
This is still open, can it be merged?

File: scripts-dev/nuke-room-from-db.sh [hunk 1]
```diff
@@ -29,7 +43,6 @@ DELETE FROM state_groups WHERE room_id = '$ROOMID';
 DELETE FROM state_groups_state WHERE room_id = '$ROOMID';
 DELETE FROM receipts_graph WHERE room_id = '$ROOMID';
 DELETE FROM receipts_linearized WHERE room_id = '$ROOMID';
-DELETE FROM event_search_content WHERE c1room_id = '$ROOMID';
```
Comment (richvdh) [inline scripts-dev/nuke-room-from-db.sh]: please could you make this the subject of a separate PR

**Comment by richvdh** [comment]
sorry for sitting on this for so long.

I would suggest that, instead of having the commandline for your database (and your database name) hardcoded in the script, it should instead write the sql to be run to stdout, and then the user can pipe it into the database. So for instance you might run

```
nuke-room-from-db.sh <room_id> | sqlite3 homeserver.db
```

or

```
nuke-room-from-db.sh <room_id> | psql --dbname=synapse
```

**Comment by rubo77** [comment]
i changed it accordingly

**Comment by rubo77** [comment]
If you fix the nuke room script, please also solve this related issue: https://github.com/matrix-org/synapse/issues/3187

**Comment by richvdh** [comment]
thanks! please could you sign off as per CONTRIBUTING.rst

**Comment by rubo77** [comment]
Signed-off-by: Ruben Barkow <github@r.z11.de>
